libs/platform/experience/composition/src/composition.component.ts

```diff
@@ -35,6 +36,26 @@ export class CompositionComponent extends LayoutMixin(
   protected registryService = resolve(ComponentsRegistryService);
   protected layoutBuilder = resolve(LayoutBuilder);
 
+  protected $dynamicVisibilityState = signal(
+    this.contentController.dynamicVisibilityState()
+  );
```

Comment:
What about `$visible` and `this.contentController.isVisible()` ?

Reply:
It is not a boolean, tt returns 4 types of states: none, defer, visible, hidden

none - when component does not have visibility rules
defer - when it has, but the value is not resolved yet
visible, hidden - when value is resolved

Reply:
either way, i'd consider dropping `dynamic` and `state` as to me those words can be added to so much things. All is dynamic, a lot of stuff is state.